Update PROJECT in accordance with operator-sdk 1.23.0

Jaanki opened the pull request:
to use kubebuilder v3. This is in line with the config new operator-sdk
framework autogenerates as evident from

https://github.com/Jaanki/submariner-operator/pull/10/files#diff-d1fb402f1269ba7396dca9cbd03b44ac055874669bce1d0f239fa2f54ea3584b

This also requires updating operator-sdk and kustomize versions. Other
changes made are:

1. Install KUSTOMIZE only if it is not locally present using the install
   script provided at its repo. Remove our own provided script.
2. Install controller tools only if it is not locally present.
3. Add make targets to install kustomize, operator-sdk and controller
   tools. This will allow users to install these versions easily.
4. Don't create packagemanifests as bundle is the recommended way to
   ship the operator.

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

Jaanki commented:
Need to update operator-sdk and kustomize versions.

tpantelis commented on Makefile:
```diff
@@ -17,10 +17,10 @@ endif
 OPERATOR_SDK := $(CURDIR)/bin/operator-sdk
 
-KUSTOMIZE_VERSION := 3.10.0
+KUSTOMIZE_VERSION := 3.8.7
```
Why was this version downgraded?

Jaanki committed:
Update PROJECT in accordance with operator-sdk 1.23.0

to use kubebuilder v3. This is in line with the config new operator-sdk
framework autogenerates as evident from

https://github.com/Jaanki/submariner-operator/pull/10/files#diff-d1fb402f1269ba7396dca9cbd03b44ac055874669bce1d0f239fa2f54ea3584b

This also requires updating operator-sdk and kustomize versions. Other
changes made are:

1. Install KUSTOMIZE only if it is not locally present using the install
   script provided at its repo. Remove our own provided script.
2. Install controller tools only if it is not locally present.
3. Add make targets to install kustomize, operator-sdk and controller
   tools. This will allow users to install these versions easily.
4. Don't create packagemanifests as bundle is the recommended way to
   ship the operator.

Part of epic: https://github.com/submariner-io/enhancements/issues/81

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

tpantelis commented on Makefile:
```diff
@@ -154,11 +154,17 @@ bin/%/submariner-operator: $(VENDOR_MODULES) main.go $(EMBEDDED_YAMLS)
-	$(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
+	test -s $(CURDIR)/bin/controller-gen || $(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
+	## TODO (Jaanki) Use go install instead
+	# $(GO) install sigs.k8s.io/controller-tools/cmd/controller-gen@v$(CONTROLLER_TOOLS_VERSION)
```
Will this TODO be resolved in this PR? If not, what's preventing us from using `go install`?

Jaanki replied:
Using `go install` fails with `go: [sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.2](http://sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.2): cannot query module due to -mod=vendor` error. I am not sure whats happening here. This is a non-blocker and should not block this PR.

tpantelis commented on Makefile:
```diff
@@ -233,25 +231,21 @@ golangci-lint: $(EMBEDDED_YAMLS)
+	## TODO (Jaanki) Add checksums
+	#gpg --verify checksums.txt.asc checksums.txt
+	#gpgv --keyring scripts/operator-sdk-signing-keyring.gpg $@.asc $@
+	#sha256sum -c scripts/operator-sdk.sha256
```
Will this TODO be resolved in this PR? If not, what's preventing it?

Jaanki replied:
I am not familiar with checksum tests using `checksum.txt` and `checksum.txt.asc`. @skitt is aware but he is on PTO. I suggest to not block the PR for this.

skitt commented on Makefile:
```diff
@@ -154,11 +136,17 @@ bin/%/submariner-operator: $(VENDOR_MODULES) main.go $(EMBEDDED_YAMLS)
+# Download controller-gen locally if not already downloaded.
+CONTROLLER_TOOLS_VERSION := 0.9.2
 $(CONTROLLER_GEN): $(VENDOR_MODULES)
 	mkdir -p $(@D)
-	$(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
+	test -s $(CURDIR)/bin/controller-gen || $(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
```
If `controller-gen` should be built only if it doesn’t exist, the prerequisites should be updated; testing for the existence of the binary is what Make does.

```suggestion
$(CONTROLLER_GEN):
	mkdir -p $(@D)
	$(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
```

skitt commented on Makefile:
```diff
@@ -182,11 +170,13 @@ is-semantic-version:
 	mkdir -p $(@D)
-	#GOBIN=$(CURDIR)/bin GO111MODULE=on $(GO) get sigs.k8s.io/kustomize/kustomize/v3
-	scripts/kustomize/install_kustomize.sh $(KUSTOMIZE_VERSION) $(CURDIR)/bin
+	test -s $(CURDIR)/bin/kustomize || { curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(CURDIR)/bin; }
```
Again, Make does this for us:

```suggestion
	{ curl -s $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(@D); }
```

skitt commented on Makefile:
```diff
@@ -233,25 +213,21 @@ golangci-lint: $(EMBEDDED_YAMLS)
-	curl -Lo $@.asc "https://github.com/operator-framework/operator-sdk/releases/download/v${OPERATOR_SDK_VERSION}/operator-sdk-v${OPERATOR_SDK_VERSION}-x86_64-linux-gnu.asc"
-	gpgv --keyring scripts/operator-sdk-signing-keyring.gpg $@.asc $@
-	sha256sum -c scripts/operator-sdk.sha256
+	mkdir -p bin && \
+	cd bin && \
```
Don’t hard-code the directory:

```suggestion
	mkdir -p $(@D) && \
	cd $(@D) && \
```
